cg/apps/lims/api.py

```diff
@@ -219,72 +235,110 @@ def update_sample(self, lims_id: str, sex=None, application: str=None,
-                method_number = prep_artifact.parent_process.udf.get('Method document')
-                method_version = (
-                        prep_artifact.parent_process.udf.get('Method document version') or
-                        prep_artifact.parent_process.udf.get('Method document versio')
-                )
-                break
 
-        if method_number is None:
-            return None
+        step_names_udfs = MASTER_STEPS_UDFS['prep_method_step']
 
-        method_name = AM_METHODS.get(method_number)
-        return f"{method_number}:{method_version} - {method_name}"
+        return self.get_methods(step_names_udfs, lims_id)
 
     def get_sequencing_method(self, lims_id: str) -> str:
         """Get the sequencing method."""
-        process_names = [
-            'CG002 - Cluster Generation (HiSeq X)',
-            'CG002 - Cluster Generation (Illumina SBS)',
-        ]
-        method_number = method_version = None
-        for process_name in process_names:
-            arts = self.get_artifacts(process_type=process_name, samplelimsid=lims_id)
-            if arts:
-                prep_artifact = arts[0]
-                method_number = (
-                        prep_artifact.parent_process.udf.get('Method') or
-                        prep_artifact.parent_process.udf.get('Method Document 1')
-                )
-                method_version = (
-                        prep_artifact.parent_process.udf.get('Version') or
-                        prep_artifact.parent_process.udf.get('Document 1 Version')
-                )
-                break
 
-        if method_number is None:
-            return None
+        step_names_udfs = MASTER_STEPS_UDFS['get_sequencing_method_step']
 
-        method_name = AM_METHODS.get(method_number)
-        return f"{method_number}:{method_version} - {method_name}"
+        return self.get_methods(step_names_udfs, lims_id)
 
     def get_delivery_method(self, lims_id: str) -> str:
         """Get the delivery method."""
-        process_name = 'CG002 - Delivery'
-        arts = self.get_artifacts(process_type=process_name, samplelimsid=lims_id)
-        if arts:
-            prep_artifact = arts[0]
-            method_number = prep_artifact.parent_process.udf.get('Method Document')
-            method_version = prep_artifact.parent_process.udf.get('Method Version')
-        else:
-            return None
 
-        method_name = AM_METHODS.get(method_number)
-        return f"{method_number}:{method_version} - {method_name}"
+        step_names_udfs = MASTER_STEPS_UDFS['get_delivery_method_step']
+
+        return self.get_methods(step_names_udfs, lims_id)
 
     def get_processing_time(self, lims_id):
         received_at = self.get_received_date(lims_id)
         delivery_date = self.get_delivery_date(lims_id)
         if received_at and delivery_date:
             return delivery_date - received_at
+
+    @staticmethod
+    def sort_by_date_run(sort_list: list):
```

Comment:
I suggest we rename the new methods to start with underscore '_' to show that they are not intendended to be part of the public API (unless the intention is to be public :)

Reply:
I agree, fixed in https://github.com/Clinical-Genomics/cg/pull/376/commits/821763d9025431ba678c41803befe8120a384676